### PR TITLE
Use golang 1.16

### PR DESCRIPTION
**Pull request by wallyworld**
Change the version of Go to 1.16

This is simply the first step - the Jenkins infrastructure also needs to be updated since the merge check tests etc still use 1.14

## QA steps

Unit tests

### CODE REVIEW

**Comment by wallyworld** [comment]
[juju core uses golang 1.16](https://trello.com/c/6DbcJ3WR/1397-juju-core-uses-golang-116)

File: go.sum [hunk 1]
```diff
@@ -1076,7 +1017,6 @@ gopkg.in/check.v1 v1.0.0-20160105164936-4f90aeace3a2/go.mod h1:Co6ibVJAznAaIkqp8
 gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
 gopkg.in/check.v1 v1.0.0-20190902080502-41f04d3bba15/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
 gopkg.in/check.v1 v1.0.0-20200227125254-8fa46927fb4f/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
-gopkg.in/check.v1 v1.0.0-20200902074654-038fdea0a05b h1:QRR6H1YWRnHb4Y/HeNFCTJLFVxaq6wH4YuVdsUOr75U=
 gopkg.in/check.v1 v1.0.0-20200902074654-038fdea0a05b/go.mod h1:Co6ibVJAznAaIkqp8huTwlJQCZ016jof/cbN4VW5Yz0=
```
Comment (jameinel) [inline go.sum:1020]: do you know why there are so many changes in this file? Is it just a go mod tidy?
Also, why do we have so many versions of gopkg.in/check.v1?

**Review by jameinel — APPROVED** [review]
Presumably we also need changes to the jenkins jobs that are kicking things off.
Are there any incompatibilities that we are worried about for 1.16? It sounds like it is a more compatible release (vs the tls changes, etc in the past)

File: go.sum (hunk 1 above)
Comment (hpidcock) [reply go.sum:1020]: these are all deletions, so yes, just go mod tidy

**Comment by wallyworld** [comment]
@jameinel there was one go 1.16 related compatibility change which I pushed in a second commit. The http server shutdown behaviour changed so we needed to modify the httpserver worker accordingly.

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Pull request closed without merging**